Feature/add ability send comments

IgorAvshtol opened the pull request:
#What has been added:
-sendData function,;
-add interceptor for axios.instance;

#What has been fixed:
-UI bug in Modal component;

IgorAvshtol committed:
feature: add Sidebar, Posts component;

IgorAvshtol committed:
feature: add useFetch custom hook;

IgorAvshtol committed:
fix: UI in Sidebar, Posts, Header;
feat: create function for getting tags from server;

IgorAvshtol committed:
feat: add favourite count;

IgorAvshtol committed:
feat: add response in UI;

IgorAvshtol committed:
fix: made it more versatile useFetch hook and added the  ability to receive data from a hook using url and generic type;
 moved the request logic to a services folder;

IgorAvshtol committed:
add: nanoid package, enum for loading status, interfaces.ts;

IgorAvshtol committed:
add: ArticlePage;

IgorAvshtol committed:
feature: add Image.tsx

IgorAvshtol committed:
fix: renamed components in Article folder;

IgorAvshtol committed:
fix: add pseudo-classes for some buttons;

IgorAvshtol committed:
fix: fixed bug display the number of likes on scroll;

IgorAvshtol committed:
fix: add ability pass additional parameters in useFetch hook;
delete: filterArticles function;

IgorAvshtol committed:
fix: add StrictMode in index.tsx;

IgorAvshtol committed:
fix: useFetch custom hook;
feat: add response UI in Article page;

IgorAvshtol committed:
feat: add Comments.tsx;

IgorAvshtol committed:
feat: add date-fns package;
delete: services folder;

IgorAvshtol committed:
feat: add sendData in useFetch;

IgorAvshtol committed:
feat: add ability get articles for all users;

IgorAvshtol committed:
feat: add test for SignUp/SignIn, Input components;

IgorAvshtol committed:
delete console.log;

IgorAvshtol committed:
fix: renamed IS_RESOLVE on IS_RESOLVED;

IgorAvshtol committed:
fix: renamed IS_RESOLVE on IS_RESOLVED;

IgorAvshtol committed:
Merge branch 'feature/Add_article_commements' into feature/Add_ability_send_comments

IgorAvshtol committed:
add: interceptor in axios.instance;

IgorAvshtol committed:
feat: add sendData function, add interceptor for axios.instance;

IgorAvshtol committed:
feat: add generic type in sendData functions;

IgorShcherba commented on src/services/dataFromStorageIsValid.ts:
```diff
@@ -0,0 +1,14 @@
+export function dataFromStorageIsValid(): IResponseData | null {
+  const data = getUserFromLocalStorage();
+  if (data) {
+    try {
```
Just move try catch into the getUserFromLocalStorage.

IgorShcherba commented on src/hooks/useProvideAuth.tsx:
```diff
@@ -55,30 +55,10 @@ export const useAuth = () => {
-  });
+  const [user, setUser] = useState<IResponseData | null>(dataFromStorageIsValid);
   const [error, setError] = useState<string>('');
   const [loading, setLoading] = useState<boolean>(false);
```
wouldn't the type be inferred here?do we need to specify it for `useState`  explicitly when it's just a primitive type

IgorAvshtol committed:
fix: bugs in UI;
feat: textarea in Sidebar on comments page, add color change Header on scroll;

IgorAvshtol committed:
Add useSWR;

IgorAvshtol committed:
feat: add useSWR, add absolute imports;
fix: bugs in UI;

IgorAvshtol committed:
delete: excess services;

IgorAvshtol committed:
fix: add margin in Sidebar;

IgorAvshtol committed:
fix: implemented auto-height for textarea using useref and useeffect;

IgorAvshtol committed:
Merge pull request #18 from IgorAvshtol/feature/Add_SWR

Feature/add swr

IgorAvshtol committed:
fix: implemented auto-height for textarea using useref and useeffect;

IgorAvshtol committed:
fix: UI;

IgorAvshtol committed:
feat: add form for new article;
fix: UI Main page, Header, Navbar;

IgorAvshtol committed:
fix: delete error field for add tags input;

IgorAvshtol committed:
fix: UI bugs;
feat: add animation in main Header, add useOneClickOutside;
delete: dataFromStorageIsValid;

IgorAvshtol committed:
fix: forbidden non-null in Animation.tsx;

IgorAvshtol committed:
fix: update UI Title, Header components;
feat: add PrivateRoute;

IgorAvshtol committed:
fix: update z-index in Header and button of Title;

IgorAvshtol committed:
fix: fixed animation on Header;

IgorAvshtol committed:
fix: add custom hook useOnClickOutside, renamed sendComment from createComment;

IgorAvshtol committed:
Merge pull request #19 from IgorAvshtol/feature/Add_ability_delete_comment

fix: implemented auto-height for textarea using useref and useeffect;

IgorAvshtol committed:
fix: remake dataFromStorageIsValid into getUserFromLocalStorage;

IgorAvshtol committed:
fix: rename sendComment into createComment;

IgorAvshtol committed:
fix: rename sendComment into createComment;

IgorAvshtol committed:
fix: rename ICreateMenu into ISendComment, add z-index 30 to Modal component;

IgorAvshtol committed:
fix: createComment in useComments;

IgorAvshtol committed:
Merge pull request #20 from IgorAvshtol/feature/Add_new_article

Feature/add new article
